src/Text/Pandoc/Writers/Typst.hs

```diff
@@ -261,10 +279,13 @@ blockToTypst block =
                           $$ ")" $$ lab $$ blankline
     Div (ident,_,_) (Header lev ("",cls,kvs) ils:rest) ->
       blocksToTypst (Header lev (ident,cls,kvs) ils:rest)
-    Div (ident,_,_) blocks -> do
+    Div (ident,_,kvs) blocks -> do
       let lab = toLabel FreestandingLabel ident
+      let props = case formatAttrs $ pickTypstAttrs kvs of
+                    [] -> ""
+                    tkvs -> parens $ literal (T.intercalate ", " tkvs)
       contents <- blocksToTypst blocks
-      return $ "#block[" $$ contents $$ ("]" <+> lab)
+      return $ "#block" <> props <> "[" $$ contents $$ ("]" <+> lab)
```

Comment:
here it's more idiomatic to use the doclayout function that puts things in brackets.